test/asyncio_tests/test_asyncio_client.py

```diff
@@ -214,6 +220,44 @@ async def test_list_database_names(self):
         self.assertIsInstance(names, list)
         self.assertIn(self.collection.database.name, names)
 
+    @unittest.skipIf(not contextvars, 'this test requires contextvars')
+    @asyncio_test
+    async def test_contextvars_support(self):
+        contextvar = contextvars.ContextVar('variable', default='default')
+
+        class Listener(monitoring.CommandListener):
+            def __init__(self):
+                self.vars = []
+
+            def save_contextvar_value(self):
+                self.vars.append(contextvar.get())
+
+            def started(self, event):
+                self.save_contextvar_value()
+
+            def succeeded(self, event):
+                self.save_contextvar_value()
+
+            def failed(self, event):
+                self.save_contextvar_value()
+
+        listener = Listener()
+        client = self.asyncio_client(event_listeners=[listener])
+        coll = client[self.db.name].test
+
+        await coll.insert_one({})
+        self.assertTrue(listener.vars)
+        for var in listener.vars:
+            self.assertEqual(var, 'default')
+
+        contextvar.set('ContextVar value')
```

Comment:
What exactly does this test?

Reply:
See below where we assert that the listener callbacks have access to the new value. This is the part that tests contextvars get forwarded to the background executors.